src/main/java/org/genomicsdb/importer/extensions/CallSetMapExtensions.java

```diff
@@ -210,6 +217,57 @@
         return generateSortedCallSetMap(listOfSampleNames, useSamplesInOrderProvided, lbRowIdx);
     }
 
+    /**
+     * Merges incremental import's callsets with existing callsets
+     * @param callsetMapJSONFilePath path to existing callset map file
+     * @param inputSampleNameToPath    map from sample name to VCF/BCF file path
+     * @param newCallsetMapPB callset mapping protobuf for new callsets
+     * @return merged callsets for callsets to TileDB rows
+     * @throws ParseException when there is an error parsing callset jsons
+     */
+    default GenomicsDBCallsetsMapProto.CallsetMappingPB mergeCallsetsForIncrementalImport(
+            final String callsetMapJSONFilePath,
+            final Map<String, URI> inputSampleNameToPath,
+            final GenomicsDBCallsetsMapProto.CallsetMappingPB newCallsetMapPB) 
+            throws ParseException {
+        String existingCallsetsJSON = GenomicsDBUtils.readEntireFile(callsetMapJSONFilePath);
+        GenomicsDBCallsetsMapProto.CallsetMappingPB.Builder callsetMapBuilder =
+                newCallsetMapPB.toBuilder();
+        checkDuplicateCallsetsForIncrementalImport(existingCallsetsJSON, inputSampleNameToPath);
+        merge(existingCallsetsJSON, callsetMapBuilder);
+        return callsetMapBuilder.build();
+    }
+
+    /**
+     * Checks for duplicates between existing and incremental callsets
+     * @param existingCallsetsJSON json string with existing callset
+     * @param inputSampleNameToPath    map from sample name to VCF/BCF file path
+     * @throws ParseException when there is an error parsing callset jsons
+     * @throws GenomicsDBException when duplicate samples are found between existing
+     * and new callsets
+     */
+    default void checkDuplicateCallsetsForIncrementalImport(
+            final String existingCallsetsJSON,
+            final Map<String, URI> inputSampleNameToPath) 
+            throws ParseException, GenomicsDBException {
+        // create PB from existing json string to iterate through it
+        // maybe better to use some json library here since we just need to 
+        // iterate through the existing callset
+        GenomicsDBCallsetsMapProto.CallsetMappingPB.Builder callsetMapBuilder =
+                GenomicsDBCallsetsMapProto.CallsetMappingPB.newBuilder();
+        merge(existingCallsetsJSON, callsetMapBuilder);
+        GenomicsDBCallsetsMapProto.CallsetMappingPB existingCallsetsPB = callsetMapBuilder.build();
+        int callsetsCount = existingCallsetsPB.getCallsetsCount();
+        for (int i=0; i<callsetsCount; i++) {
+            String sampleName = existingCallsetsPB.getCallsets(i).getSampleName();
+            URI value = inputSampleNameToPath.get(sampleName);
+            if (value != null) {
+                throw new GenomicsDBException("Duplicate sample name found: "+sampleName+". Sample "+
+                        "was originally in "+value);
+            }
```

Comment:
It will be useful to get a list of all duplicates before throwing the exception.

Reply:
I'm following what GATK does here for duplicates within an import - they throw exception on the first duplicate. Do you think a (potentially long) list of duplicates would be useful to the users?

Reply:
We could stop at some predetermined number. But, only if this feature is useful in the first place. Your call.